### DISCUSSION TITLE
Get the time a commit was pushed?

### DISCUSSION THREAD

**Discussion by hatboysam** [community/community#24730] [Apps, API and Webhooks]
<p>Using this API I can list the commits on a Pull Request:<br>
<a href="https://docs.github.com/en/rest/reference/pulls#list-commits-on-a-pull-request" class="onebox" target="_blank" rel="noopener nofollow ugc">https://docs.github.com/en/rest/reference/pulls#list-commits-on-a-pull-request</a></p>
<p>As far as I understand, the <code>commit.committer.date</code> and <code>commit.author.date</code> fields are the time the user ran <code>git commit</code> locally and not when they did the <code>git push</code> which caused the Pull Request to update.</p>
<p>However when I look at a Pull Request on GItHub I can see information on push times:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/b/cbe16ce4fdb2629eea40cece6a7747ac2ce78007.png" data-download-href="/uploads/short-url/t5BTGoPk1BfVxhNIDlwD1KtrBD9.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/997280/181086487-1263b856-5799-433b-88de-1d6e4698fd7a.png" alt="image" data-base62-sha1="t5BTGoPk1BfVxhNIDlwD1KtrBD9" width="690" height="94" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/b/cbe16ce4fdb2629eea40cece6a7747ac2ce78007_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">887×122 6.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Is there some way to get this information via the API to reconstruct the PR timeline from the perspective of the reviewer?</p>

**Comment by jsoref** [comment]
<p>It’s available from the repository’s events api:<br>
<a href="https://api.github.com/repos/Codertocat/Hello-World/events" class="onebox" target="_blank" rel="noopener nofollow ugc">https://api.github.com/repos/Codertocat/Hello-World/events</a></p>
<p>You might be able to find an equivalent in GraphQL as well:</p><aside class="quote" data-post="1" data-topic="13600">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/get-event-equivalent-for-v4/13600">GET /event equivalent for v4?</a> <a class="badge-wrapper  bullet" href="/c/github-ecosystem/37"><span class="badge-category-bg" style="background-color: #28a745;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and support using GitHub’s REST, and GraphQL APIs, building Applications and OAuth Apps, the GitHub Marketplace, and third party integrations.">GitHub Ecosystem</span></a>
  </div>
  <blockquote>
    So, I’m trying to switch my small project from REST to GraphQL. Looks nice to fetch exactly what’s needed. 
But what I’m mostly fetching from REST are <a href="https://developer.github.com/v3/activity/events/#events" rel="noopener nofollow ugc">events</a> from v3 but i can’t find it on the v4-docs. 
Is there any equivalent to events on GraphQL?
  </blockquote>
</aside>

**Comment by hatboysam** [comment]
<p>Thanks! Can the events API be queried by event type or filtered in any meaningful way? It doesn’t seem efficient to page through all repo events until I find the particular SHA I am looking for.</p>

**Comment by jsoref** [comment]
<p>Dunno. In general if you want filtering you’re supposed to use GraphQL.</p>
<p>I’m not a GraphQL expert. It’s sometimes powerful, although its power varies significantly depending on how the APIs are written.</p>
<p>I practice, I’m definitely a pessimist.</p>
<p>If I were you, I’d contact:</p>
<p><a href="https://support.github.com" class="onebox" target="_blank" rel="noopener nofollow ugc">https://support.github.com</a></p>
<p>And describe what you’re trying to do, having looked through both APIs and determined that they do not have an efficient way of retrieving the necessary information (obviously you get to do this step first).</p>

**Comment by hatboysam** [comment, marked as answer]
<p>Doesn’t look like this is possible in the REST API but it is possible using GQL, for example:</p>
<pre><code class="lang-auto">{
      repository(owner: "firebase", name: "firebase-tools") {
        pullRequest(number: 3544) {
          timelineItems(last: 250, itemTypes: [PULL_REQUEST_COMMIT]) {
            nodes {
              ... on PullRequestCommit {
                commit {
                  oid,
                  committedDate,
                  pushedDate
                }
              }
            }
          }
        }
      }
    }
</code></pre>
<p><code>pushedDate</code> will be <code>null</code> in the case where the PR is from a fork, it looks like that information is just not available in that case.</p>

**Reply by dlowe** [reply]
n.b. `pushedDate` was deprecated in 2023; I don't think this is a working answer anymore :(

**Comment by jsoref** [comment]
<p>A given commit can be pushed to multiple (related) repositories at different time points.</p>
<p>Since a PR from a related PR likely contains a commit that has yet to be pushed into your repository, it’s correct to have a null <code>pushedDate</code> from your repository’s perspective.</p>
<p>If you asked the repository where the commit id does exist, GitHub should be able to tell you when it was pushed into <em>that</em> repository.</p>

**Comment by hatboysam** [comment]
<p>Ah I see, thanks! That gets a bit tricky with permissions I assume but good to know.</p>

**Comment by jsoref** [comment]
<p>I don’t think you’ll run into permissions. If a repository is in your fork network, then you can see it, and if you can see it, then I’m pretty sure you can see when the commit was pushed.</p>
<p>I could be wrong.</p>
<p>You will of course have to pay for extra API hits. But you could construct a GQL query that combines a bunch of those lookups.</p>

**Comment by cirosantilli** [comment]
For finding a specific commit I'd also have a look at: https://www.gharchive.org a bit more chatter at: https://stackoverflow.com/a/78966680/895245 Not sure it's actually possible there though.
